src/manager/kiwix-manage.cpp

```diff
@@ -96,24 +96,24 @@ void usage()
             << std::endl;
 }
 
-int handle_show(kiwix::Library* library, const std::string& libraryPath,
+int handle_show(kiwix::ConstLibraryPtr library, const std::string& libraryPath,
                  int argc, char* argv[])
 {
   if (argc > 3 ) {
     for(auto i=3; i<argc; i++) {
        std::string bookId = argv[i];
-       show(library, bookId);
+       show(*library, bookId);
     }
   } else {
     auto booksIds = library->getBooksIds();
     for(auto& bookId: booksIds) {
-      show(library, bookId);
+      show(*library, bookId);
     }
   }
   return(0);
 }
 
-int handle_add(kiwix::Library* library, const std::string& libraryPath,
+int handle_add(kiwix::LibraryPtr library, const std::string& libraryPath,
```

Comment:
I think that a parameter of type `kiwix::Library&` is more suitable here (passing a `shared_ptr` suggests that it is preserved somewhere and can be accessed after the function returns).

Reply:
We pass the library to the `Manager` which take a `LibraryPtr`.
So I have keep `LibraryPtr` instead of `kiwix::Library&`.

Reply:
Since the `Manager` object is completely internal to `handle_add()` (it no longer exists when the function returns) it would be perfectly safe to convert the `Library&` into a non-owning `shared_ptr` inside the function.